ios/MattermostShare/ShareViewController.swift

```diff
@@ -42,9 +42,13 @@ class ShareViewController: SLComposeServiceViewController {
   }
   
   override func isContentValid() -> Bool {
-    // Do validation of contentText and/or NSExtensionContext attachments here
-    if (attachments.count > 0) {
+    let maxMessageSize = store.getMaxPostSize()
+    //Check content text size is not above max
+    if (contentText.count > maxMessageSize) {
+      showErrorMessage(title: "", message: "Content text shared in Mattermost must be less than \(maxMessageSize+1) characters.", VC: self)
```

Comment:
I think even in case you are sharing a long text this message should only be shown if the user is logged in, if the user is not logged in should display the other message

Reply:
There are two alert messages here.  One message is if the post size is too big and one message is if the attachment size is too big.  If you aren't logged in, another message is shown, but this is already included in the existing, unmodified functionality.  I've included a screenshot of what currently occurs if you attempt to use the iOS social sharing feature while not logged in:

![Screen Shot 2019-04-15 at 10 20 22 AM](https://user-images.githubusercontent.com/10441971/56152381-5e9b8a00-5f68-11e9-8db7-ea55975a1dc0.png)